src/main/java/gyro/google/compute/InstanceResource.java

```diff
@@ -0,0 +1,316 @@
+        content.setDescription(getDescription());
+        content.setMachineType(getMachineType());
+        content.setNetworkInterfaces(getNetworkInterfaces().stream()
+            .map(InstanceNetworkInterface::copyTo)
+            .collect(Collectors.toList()));
+        content.setLabels(getLabels());
+        content.setDisks(getInitializeDisks().stream()
+            .map(InstanceAttachedDisk::copyTo)
+            .collect(Collectors.toList()));
+
+        waitForCompletion(
+            client,
+            client.instances().insert(getProjectId(), getZone(), content).execute());
+
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        Compute client = createComputeClient();
+
+        if (changedFieldNames.contains("labels")) {
+            waitForCompletion(
+                client,
+                client.instances()
+                    .setLabels(
+                        getProjectId(),
+                        getZone(),
+                        getName(),
+                        new InstancesSetLabelsRequest()
+                            .setLabelFingerprint(getLabelFingerprint())
+                            .setLabels(getLabels()))
+                    .execute());
+        }
+
+        refresh();
+    }
+
+    @Override
+    public void doDelete(GyroUI ui, State state) throws Exception {
+        Compute client = createComputeClient();
+        client.instances().delete(getProjectId(), getZone(), getName()).execute();
+    }
+
+    @Override
+    public void copyFrom(Instance model) {
+        setName(model.getName());
+        setZone(model.getZone().substring(model.getZone().lastIndexOf("/") + 1));
+        setDescription(model.getDescription());
+        setMachineType(model.getMachineType());
+        setSelfLink(model.getSelfLink());
+        setLabelFingerprint(model.getLabelFingerprint());
+        setLabels(model.getLabels());
+
+        getNetworkInterfaces().clear();
+        if (model.getNetworkInterfaces() != null) {
+            setNetworkInterfaces(model.getNetworkInterfaces().stream()
+                .map(networkInterface -> {
+                    InstanceNetworkInterface newNetworkInterface = newSubresource(InstanceNetworkInterface.class);
+                    newNetworkInterface.copyFrom(networkInterface);
+                    return newNetworkInterface;
+                })
+                .collect(Collectors.toList()));
+        }
+
+        getDisks().clear();
+        if (model.getDisks() != null) {
+            setDisks(model.getDisks().stream()
+                .map(disk -> {
+                    InstanceAttachedDisk instanceAttachedDisk = newSubresource(InstanceAttachedDisk.class);
+                    instanceAttachedDisk.copyFrom(disk);
+                    return instanceAttachedDisk;
+                })
+                .collect(Collectors.toList()));
```

Comment:
we use findById to find the resource already present in the config

Reply:
This is a sub resource. Keeping.